src/utils.py

```diff
@@ -59,6 +60,48 @@ def do_build_all_images():
             and JobParameters.py_versions == constants.ALL
         )
 
+def download_s3_file(bucket_name, filepath, local_file_name):
+    """
+
+    :param bucket_name: string
+    :param filepath: string
+    :param local_file_name: string
+    :return:
+    """
+    _s3 = boto3.Session().resource('s3')
+
+    try:
+        _s3.Bucket(bucket_name).download_file(filepath, local_file_name)
+    except ClientError as e:
+        LOGGER.error("Error: Cannot read file from s3 bucket.")
+        LOGGER.error("Exception: {}".format(e))
+        raise
+
+def download_file(remote_url: str, link_type: str):
+    """
+    Fetch remote files and save with provided local_path name
+    :param link_type: string
+    :param remote_url: string
+    :return: file_name: string
+    """
+    LOGGER.info(f"Downloading {remote_url}")
+
+    ctx = Context()
+    file_name = ctx.run(f"basename {remote_url}").stdout.strip()
+    LOGGER.info(f"basename: {file_name}")
```

Comment:
Why does file_name need to be identified through `ctx.run`? Why not use os.path.basename()?

Reply:
Thanks, corrected to using os.path.basename.